Funcky/Extensions/EnumerableExtensions/MaxByOrNone.cs

```diff
@@ -0,0 +1,47 @@
+namespace Funcky.Extensions;
+
+public static partial class EnumerableExtensions
+{
+    /// <summary>Returns the maximum value in a generic sequence according to a specified key selector function.</summary>
+    /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
+    /// <typeparam name="TKey">The type of key to compare elements by.</typeparam>
+    /// <param name="source">A sequence of values to determine the maximum value of.</param>
+    /// <param name="keySelector">A function to extract the key for each element.</param>
+    /// <returns>The value with the maximum key in the sequence or None.</returns>
+    public static Option<TSource> MaxByOrNone<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector)
+        where TSource : notnull
+        where TKey : notnull
```

Comment:
Why is `TKey` also marked as notnull?

Reply:
Fixed in ad6ea1a9a8bba1f751caba41b6208ff2799173b5